### ISSUE TITLE
import jax fails on numpy version issue.

### ISSUE THREAD

**Issue by silverpaths** [google/jax#2606]
Hi There,

I've just done an install of jax on a fresh install of Ubuntu 18.04 LTS --- as vanilla as possible.
I followed the installation instructions for CPU+GPU support.

When I try to import jax in python I get the following:

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
Type "copyright", "credits" or "license" for more information.

IPython 5.5.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import jax
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-cb15c4215ef7> in <module>()
----> 1 import jax

/home/case/.local/lib/python3.6/site-packages/jax/__init__.py in <module>()
     17 
     18 from jax.version import __version__
---> 19 from jax.api import *
     20 from jax import nn
     21 from jax import random

/home/case/.local/lib/python3.6/site-packages/jax/api.py in <module>()
     35 from contextlib import contextmanager
     36 
---> 37 from . import core
     38 from . import linear_util as lu
     39 from . import ad_util

/home/case/.local/lib/python3.6/site-packages/jax/core.py in <module>()
     28 import numpy as onp
     29 
---> 30 from . import dtypes
     31 from . import linear_util as lu
     32 

/home/case/.local/lib/python3.6/site-packages/jax/dtypes.py in <module>()
     29 from . import util
     30 from .config import flags
---> 31 from .lib import xla_client
     32 
     33 FLAGS = flags.FLAGS

/home/case/.local/lib/python3.6/site-packages/jax/lib/__init__.py in <module>()
     50 except:
     51   tpu_client = None
---> 52 from jaxlib import xla_client
     53 from jaxlib import lapack
     54 

/home/case/.local/lib/python3.6/site-packages/jaxlib/xla_client.py in <module>()
     35 # and TensorFlow may fail with duplicate protocol buffer message definitions.
     36 
---> 37 from . import xla_extension as _xla
     38 from .xla_extension import ops
     39 

ImportError: Unable to initialize Numpy API

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

I've tried searching for this issue and looking through these archives, but can't find anyone with this problem. 
Any idea where I should start with trying to figure out the install? 
As mentioned, I tried to keep everything as vanilla as possible: fresh Ubuntu 18.04 install, installed nvidia drivers with no issue, installed cuda 10.1 with no issue, the install instructions appeared to worked for jax.
So it seems quite possible that others may run into the issue.

Any help appreciated.

**Comment by shoyer** [comment]
Can you install a newer version of NumPy? That error is reporting that you have NumPy 1.12 installed, but JAX was compiled against NumPy 1.13. Any newer version of NumPy should work fine .

**Comment by silverpaths** [comment]
Thanks, that sent me down the right path and I managed to fix the issue. 

I've included what did below in case someone has a similar issue.
There was something quite strange going on because I did have NumPy 1.13 installed, and that's what was being loaded by default by my python interpreter when I imported numpy:
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
In [7]: import numpy as np
In [8]: np.__version__
Out[8]: '1.13.3'
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

So perhaps jax was somehow finding a different version of numpy somewhere?

Anyway, just to be safe, I removed my NumPy via:
sudo apt remove --purge python3-numpy

And installed it via pip instead:
pip3 install --user numpy

And now everything works (though the NumPy version fetched by pip is actually 1.18.2):
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
In [1]: import numpy as np
In [2]: np.__version__
Out[2]: '1.18.2'
In [3]: import jax
In [4]: 
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

Thanks again!
